File: src/Blog/Blog.Blazor/Data/Entities/ExperimentTranslation.cs
using Blog.Blazor.Data;

namespace Blog.Blazor.Data.Entities;

public sealed class ExperimentTranslation
{
    public int Id { get; set; }
    public int ExperimentId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Purpose { get; set; } = string.Empty;
    public string? Body { get; set; }

    [JsonOrEmpty]
    public string? Seo { get; set; }

    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Experiment? Experiment { get; set; }
}

File: src/laravel/app/Models/ExperimentTranslation.php
<?php

namespace App\Models;

use App\Models\Concerns\Auditable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ExperimentTranslation extends Model
{
    use Auditable, HasFactory;

    protected $fillable = ['experiment_id', 'locale', 'name', 'purpose', 'body', 'seo'];

    protected $casts = ['seo' => 'array'];

    /**
     * @return BelongsTo<Experiment, $this>
     */
    public function experiment(): BelongsTo
    {
        return $this->belongsTo(Experiment::class);
    }
}
